### PR TITLE
Disposals Rework [DNM]

### LINKED ISSUE #4654: Disposals Issue and features B:150
Known issues from #4524.
### Known issues

- [x] fix #6568.
- [x] 22. Items ejected from a disposal pipe terminal sometimes permanently spin, as described in issue #5551.
- [x] 3. Constructed pipes with non-standard grey colour return to default grey when welded to form a tile.
- [x] 4. Sometimes items can be ejected at a strange angle.
- [x] 13. Players are exposed to vacuum when traversing disposal pipes that are in space.
- [x] 14. Disposal machines do not use power from APCs.
- [x] 15. Disposal virtual containers can be seen by players if they bring up the context menu for the same tile the container is on. This is issue #4619.
- [x] ~~16. Comply with requested changes in #4524.~~ _**Partially addressed by PR #5786.**_
- [x] 7. ~~Dropping items while in a disposal instance does not add them to the disposal virtual container, and so they are lost to the void.~~ Items dropped while in a disposal instance are invisible once ejected.
- [x] 8. Coloured machine does not apply to overlays (bin overlay spritehandler).
- [ ] 9. Tiles aren't actually broken when items are ejected from pipe end - the update is merely visual.
- [x] 19. DisposalVirtualContainer can be spawned via the dev spawner menu.
#
### New features
- [x] G. Dump the contents of the trash bag into the disposal bin without dumping the bag itself, as described in #4938. This should probably be a property of the bag and not the bin.
- [x] H. Disposal bins should actually take in ambient air, store it, and send it with the disposal traversal. Lungs inside the instance should then breathe from this gas mix. Should solve issue part 13. in the process.
- [x] C. A tooltip for the pipe dispenser's dispensables would be nice, too.

 #
- [x] **Disposals crawling** - When a player enters a disposal pipe system, they should be able to stop flowing along the pipes by pressing the resist key. After struggling for a period of time, they will stop moving with the flow and instead be able to crawl through the pipes in any direction they want to. 

They should also be able to enter disposal pipe openings in a similar manner to disposal bins to initiate crawling at any time they would like, allowing them to navigate pipes in a similar way to small creatures vent crawling. They should also be able to crawl out of disposal pipe openings and disposal bins to get free of the system. 

Disposals crawling will greatly expand the possible applications of the disposals system and add an effective counter to being trapped in disposal pipe loops set up by griefers.

### LINKED ISSUE #6568: Disposal bins not initialising correctly
Disposal bins don't seem to be initialising correctly:
off
While the expected sprite state is:
ready
~~In addition, disposal bins spawned in without a disposal pipe terminal underneath are spawning as immediately above, and not rotated, as depicted here:
uninstalled
{spawning over a disposal pipe terminal is as expected)~~

### PR DESCRIPTION
**Pull request by dapocalypse**
### Purpose
Fixes #4654
Fixes #6568

This is a disposals rework.

Disposals bins are on at start.

### Notes:
Edit: Now disposals machines spawn as intended.

### CODE REVIEW

**Comment by NoooneyDude** [comment]
There's a problem with SpriteHandler where using PushClear to set no sprites seems a bit buggy, as it thinks the sprite hasn't changed and so doesn't set the new one. I suspect this might be the case here. There's a sprite SO that you can use as a blank, so just switch to this SO in order to clear the sprite.

Might be worth a go?

File: UnityProject/Assets/Scripts/Objects/Disposals/DisposalBin.cs [hunk 1]
```diff
@@ -151,17 +150,10 @@ protected override void UpdateSpriteConstructionState()
 
 		private void UpdateSpriteBinState()
 		{
-			if (MachineUnattached)
-			{
-				baseSpriteHandler.ChangeSprite((int) BinSprite.Sideways);
-				overlaysSpriteHandler.PushClear();
-				return;
-			}
-
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/Disposals/DisposalBin.cs:160]: Why remove this?

File: UnityProject/Assets/Scripts/Objects/Disposals/DisposalBin.cs [hunk 2]
```diff
@@ -151,17 +150,10 @@ protected override void UpdateSpriteConstructionState()
...
 			switch (binState)
 			{
 				case BinState.Disconnected:
-					baseSpriteHandler.ChangeSprite((int) BinSprite.Upright);
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/Disposals/DisposalBin.cs:164]: Disconnected should be upright. Disconnected refers to the state where you first: turn off the power using the UI, then use a screwdriver to disconnect the power cable as per /tg/.

The sideways sprite is for when the disposal bin is unwrenched and moveable.

File: UnityProject/Assets/Scripts/Objects/Disposals/DisposalBin.cs (hunk 1 above)
Comment (dapocalypse) [reply UnityProject/Assets/Scripts/Objects/Disposals/DisposalBin.cs:160]: It got stuck in the loop on roundstart. I can just add unconnected to the switch statement

**Comment by dapocalypse** [comment]
> There's a problem with SpriteHandler where using PushClear to set no sprites seems a bit buggy, as it thinks the sprite hasn't changed and so doesn't set the new one. I suspect this might be the case here. There's a sprite SO that you can use as a blank, so just switch to this SO in order to clear the sprite.
> 
> Might be worth a go?

I don't think this is the issue. The ready state isn't using PushClear its changing the sprite.

**Comment by dapocalypse** [comment]
Remove merge conflicts

**Pull request closed without merging**